harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -657,6 +657,61 @@ def test_amp(self, trial_class) -> None:
         )
         controller.run()
 
+    @pytest.mark.skipif(not HAVE_APEX, reason="Apex not available")
+    @pytest.mark.skipif(not torch.cuda.is_available(), reason="no gpu available")
+    @pytest.mark.gpu
+    def test_apex_checkpointing(self, tmp_path: pathlib.Path) -> None:
+        # There will be num_checkpoints-1 assertions made at the end of this test
+        num_checkpoints = 5
+        training_metrics = {}
+        validation_metrics = {}
+        checkpoint_dir = str(tmp_path.joinpath("apex_checkpoint"))
+        latest_checkpoint = None
+        steps_completed = 0
+
+        def make_workloads(checkpoint_idx: int = 0) -> workload.Stream:
+            # Train as usual
+            trainer = utils.TrainAndValidate()
+            # In order to expect the loss to decrease after loading the first checkpoint,
+            #  make sure the scaler has had enough steps to stop reducing so that training
+            #  can actually proceed.
+            yield from trainer.send(steps=10, validation_freq=10)
```

Comment:
nit: why even validate at all?  Validation should not affect the scaler at all.

Reply:
I would agree, but `validation_freq` isn't an optional parameter, so I used the value that minimizes the time spent validating without needing clarification (like, I didn't put `11` or `inf`).

Reply:
huh, I didn't realize we couldn't not validate.  Makes sense then.